lib/galaxy/tools/actions/__init__.py

```diff
@@ -236,7 +236,7 @@ def execute(self, tool, trans, incoming={}, return_job=False, set_output_hid=Tru
                 incoming[ "%s|__identifier__" % name ] = identifier
 
             for tag in data.tags:
-                if tag.user_tname == 'name':
+                if tag.user_tname == 'name' and tag.value not in [t.value for t in preserved_tags]:
```

Comment:
Sure, upload two files, create a collection with one file each, tag both collections with the same tag. Then concatenate (with galaxy's concat1 tool) the two collections and you will have a duplicate tag.